### PR TITLE
[DM-52] Add preview templates

### PR DESCRIPTION
**Pull request by VicThorX**
Hi team!

Here I added 2 features:
- In the list of templates, I added one button to view in another window, the generated image.
image

- And when you create or edit a template, at the bottom you can see in real-time the rendered HTML, which is taken from the template content.
screencapture-dopplerrelayqa-makingsense-2019-09-25-16_27_56

If someone wants to try it, you can go to this URL: [Relay QA](https://dopplerrelayqa.makingsense.com/)
And log in with username1@dopplerrelay.com and password1

If anyone has suggestions, please comment!

### CODE REVIEW

File: src/wwwroot/controllers/TemplatesCtrl.js [hunk 1]
```diff
@@ -50,6 +51,10 @@
         });
     }
 
+    function previewTemplate(thumbnailUrl) {
+      window.open(thumbnailUrl);
```
Comment (andresmoschini) [inline src/wwwroot/controllers/TemplatesCtrl.js]: it is weird. Now you have the HTML rendering, why showing a small image?

Comment (VicThorX) [reply src/wwwroot/controllers/TemplatesCtrl.js]: This shouldn't be here. My bad!

**Comment by andresmoschini** [comment]
❤️ Please, wait for @mvillaseco review before merge.

File: src/wwwroot/controllers/TemplateCtrl.js [hunk 2]
```diff
@@ -27,10 +28,14 @@
     $scope.loadInProgress = false;
     $scope.saveInProgress = false;
     $scope.isHtmlTemplate = true;
-
+    $scope.htmlPreviewContent = "";
     $scope.saveTemplate = saveTemplate;
     $scope.saveAndEditWithMseditor = saveAndEditWithMseditor;
 
+    $scope.$watch("template.content", function(htmlContent) {
+      $scope.htmlPreviewContent = $sce.trustAsHtml(htmlContent);
```
Comment (mvillaseco) [inline src/wwwroot/controllers/TemplateCtrl.js]: 💯

File: src/wwwroot/styles/views/_templates.scss [hunk 3]
```diff
@@ -44,6 +44,7 @@
    &:first-of-type{ margin-right: 3%;}
    &:last-of-type{ margin-left: 3%;}
   }
+  .flex{align-items: center;}
```
Comment (mvillaseco) [inline src/wwwroot/styles/views/_templates.scss]: the class flex it's a common atomic name, maybe you could rename it?

File: src/wwwroot/partials/templates/template.html [hunk 4]
```diff
@@ -95,6 +95,14 @@ <h2>{{'templates_edition_title' | translate}}</h2>
               ng-disabled="saveInProgress"
               value="{{'templates_save_button' | translate}}" />
       </div>
+      <div class="section--wrapper text--center">
+        <input type="button"
+              class="button button--medium"
+
```
Comment (cbernat) [inline src/wwwroot/partials/templates/template.html]: Could you remove this empty space?

File: src/wwwroot/styles/views/_templates.scss [hunk 5]
```diff
@@ -91,7 +92,7 @@
       @include col(1/3, $cycle:3);
       padding: 0 !important;
       transition:all linear 0.5s;
-      img{height: 180px; width: 100%;object-fit: cover;object-position: left;}
+      img{height: 180px; width: 100%;object-fit: contain;}
```
Comment (cbernat) [inline src/wwwroot/styles/views/_templates.scss]: How about spreading the class definition for clarity? Like this:
```
img{
  height: 180px; 
  width: 100%;
  object-fit: contain;
}
```

**Comment by VicThorX** [comment]
test this please

**Comment by VicThorX** [comment]
test this please

**Comment by VicThorX** [comment]
test this please

**Comment by VicThorX** [comment]
test this please

**Comment by VicThorX** [comment]
test this please

**Comment by VicThorX** [comment]
test this please

**Comment by VicThorX** [comment]
test this please

**Comment by VicThorX** [comment]
test this please

**Pull request merged**